src/databricks/labs/remorph/helpers/recon_config_utils.py

```diff
@@ -0,0 +1,186 @@
+import json
+import os
+
+from databricks.connect import DatabricksSession
+from databricks.labs.blueprint.entrypoint import get_logger
+from databricks.labs.blueprint.tui import Prompts
+from databricks.sdk import WorkspaceClient
+
+from databricks.labs.remorph.helpers.db_workspace_utils import DBWorkspaceClient
+from databricks.labs.remorph.reconcile.connectors.snowflake import SnowflakeDataSource
+from databricks.labs.remorph.reconcile.constants import SourceType
+from databricks.labs.remorph.reconcile.recon_config import TableRecon
+
+logger = get_logger(__file__)
+
+recon_source_choices = [
+    SourceType.SNOWFLAKE.value,
+    SourceType.ORACLE.value,
+    SourceType.DATABRICKS.value,
+    SourceType.NETEZZA.value,
+]
+
+
+class ReconConfigPrompts:
+    def __init__(self, ws: WorkspaceClient):
+        self._source = None
+        self._prompts = Prompts()
+        self._db_ws = DBWorkspaceClient(ws, self._prompts)
+
+    def prompt_source(self):
+        source = self._prompts.choice("Select the source", recon_source_choices)
+        self._source = source
+        return source
+
+    def prompt_snowflake_connection_details(self) -> tuple[str, dict[str, str]]:
+        logger.info(
+            f"Please answer a couple of questions to configure `{SourceType.SNOWFLAKE.value}` Connection profile"
+        )
+
+        sf_url = self._prompts.question("Enter Snowflake URL")
+        account = self._prompts.question("Enter Account Name")
+        sf_user = self._prompts.question("Enter User")
+        sf_password = self._prompts.question("Enter Password")
```

Comment:
Does the `question` method know that it should not print the user's input when asking for a password or other kind of sensitive information?

Reply:
I don't think so, it shows the sensitive information on screen. But after that, we're storing that information in configured `Databricks Secrets scope`. 
In `databricks cli` `Secrets` they have the provision to read from `json` /  `bytes value` / `string`. We will add this provision here [if needed]. 

> https://docs.databricks.com/en/dev-tools/cli/commands.html 

```
Usage:
  databricks secrets put-secret SCOPE KEY [flags]

Flags:
      --bytes-value string    If specified, value will be stored as bytes.
  -h, --help                  help for put-secret
      --json JSON             either inline JSON string or @path/to/file.json with request body (default JSON (0 bytes))
      --string-value string   If specified, note that the value will be stored in UTF-8 (MB4) form.
```

<img width="1146" alt="image" src="https://github.com/databrickslabs/remorph/assets/84904851/bd177bc2-4383-4308-868d-67c41204b1f0">

Reply:
The possibility of having sensitive information displayed on screen feels like a security liability, but I let @nfx decide whether it should prevent this PR from being merged as is.

Reply:
well... eventually we should switch to displaying the HTML page on localhost webserver, but that would come later.